src/Symfony/Component/Notifier/Notification/Notification.php

```diff
@@ -36,21 +36,19 @@ class Notification
     public const IMPORTANCE_MEDIUM = 'medium';
     public const IMPORTANCE_LOW = 'low';
 
-    private array $channels = [];
-    private string $subject = '';
```

Comment:
Must be reverted to non-CPP. These properties lose their default values with CPP.

Reply:
This might only be a problem when either overriding this class without calling the parent constructor or creating an instance of this class without calling the constructor (e.g. through reflection). Are those scenarios we want to support?

Reply:
We don't provide BC for reflecting non-public API as far as I know. However, the constructor overriding without parent constructor is something we should protect against when the class is not final.

I've mentioned this a previous time in a "move to CPP" PR and Nicolas confirmed that we already had some bug reports about this in other classes in Symfony. So I think it's better to stay on the save side and reject CPP for properties with default values on non-final classes (and we can't make DTO's final as that limits extensibility to zero).

Reply:
Not calling this constructor looks like a mistake to me, but you're right, it might've happened and we should not break code that does so.

Still, I'd like to see a path forward for such classes. Maybe that change would be acceptable for Symfony 8, although I don't really see how we could build a deprecation layer for that.

Reply:
So in short, revert all CPP on non final classes, having a default value?

Reply:
100% correct :)

And in both of you PRs, this is the only one I spotted (the other PR has one occurence of properties with default values, but it's on a final class)

Reply:
```diff
- So in short, revert all CPP on non final classes, having a default value
+ So in short, revert all CPP on non final classes, having a default value on the property
```

Reply:
> Not calling this constructor looks like a mistake to me, but you're right, it might've happened and we should not break code that does so.

I'm pretty sure we already rejected regression report(s) due to missing parent constructor calls in the past (mostly argued by @stof IIRC) so I'm mitigated, but I tend to agree we should not do it consciously for the sake of adding CPP.

Reply:
Done